### PR TITLE
[#IOPSC-113] Add IP Addresses

### PR DESCRIPTION
**Pull request by LoFrance**
We need to retrieve IP Addresses from Cosmos and save on PostgreSQL.

#### List of Changes
- Change Schema: add JSON field to represent a lists of IPs
- Change Jest config
- Change code review pipeline
- SQL: Need to add sql to take IP values from Cosmos (WIP)

#### Motivation and Context
It will be useful to have a table where we can store the authorized IP and retrieve for our checks.

#### How Has This Been Tested?
Will be some jest tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by LoFrance — COMMENTED** [review]
The insert query doesn't add the `authorizedCIDRS` data.

File: schema/migrations/db/V1__schema.sql [hunk 1]
```diff
@@ -19,5 +19,5 @@ CREATE TABLE IF NOT EXISTS "${schemaName}".services
     "subscriptionAccountEmail" text,
 
     "updateAt" time without time zone DEFAULT CURRENT_TIMESTAMP,
-    CONSTRAINT migrations_pkey PRIMARY KEY ("id")
+    CONSTRAINT servicedata_pkey PRIMARY KEY ("id")
```
Comment (balanza) [inline schema/migrations/db/V1__schema.sql:22]: Once applied, we cannot edit sql schema migrations. Instead, we should alter the current schema via a new migration step.

**Review by balanza — CHANGES_REQUESTED** [review]
It's better to deploy schema changes and code changes in separated steps to avoid inconsistencies and edge cases.

Can you split these changes in two PRs?

File: models/Domain.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { ServiceSearchItem } from "../generated/definitions/ServiceSearchItem";
 export type MigrationRowDataTable = t.TypeOf<typeof MigrationRowDataTable>;
 export const MigrationRowDataTable = t.interface({
   // data related to the subscription received from COSMOS
+  authorizedCIDRS: t.interface({ ip: t.readonlyArray(t.string) }),
```
[the remark this answers is not in the document]
Comment (LoFrance) [reply models/Domain.ts:12]: Because this is a JSON data and I need an object.

**Comment by LoFrance** [comment]
> It's better to deploy schema changes and code changes in separated steps to avoid inconsistencies and edge cases.
> 
> Can you split these changes in two PRs?

Yep, you're right, I split it in two:

- for DB Schema: https://github.com/pagopa/io-developer-portal-service-data/pull/15
- for Domain and Code: https://github.com/pagopa/io-developer-portal-service-data/pull/16

File: schema/migrations/db/V1__schema.sql (hunk 1 above)
Comment (LoFrance) [reply schema/migrations/db/V1__schema.sql:22]: Oh, I guessed to reset the DB however I've added a new v4 migration schema file to do a DROP and ADD a new constraint, unfortunately a Primary Key constraint can't be altered.

**Pull request closed without merging**